Rover/Parameters.cpp

```diff
@@ -694,6 +694,12 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Path: mode_circle.cpp
     AP_SUBGROUPINFO(mode_circle, "CIRC", 57, ParametersG2, ModeCircle),
 
+    // @Param: DRIVE_OPTIONS
```

Comment:
I'd like to avoid this new parameter if possible.  I think the user could achieve essentially the same thing by setting the INITIAL_MODE parameter to the mode that requires a position estimate?

Reply:
The point here is to require that the vehicle has position in all modes.  So the user can't arm the vehicle at all until it knows how to get home.

Reply:
yes, I definitely understand the need for this now and we have the same option in Copter.  I just wonder why we put it into FLIGHT_OPTIONS instead of ARMING_OPTIONS as part of PR https://github.com/ArduPilot/ardupilot/pull/27367

Reply:
> yes, I definitely understand the need for this now and we have the same option in Copter. I just wonder why we put it into FLIGHT_OPTIONS instead of ARMING_OPTIONS as part of PR #27367

I think we *did* discuss at DevCall - but I can't find any notes on it.

It probably should be in arming options.  Should I move it?  This will require some awkward code in Copter if we want people to start using  the bit in ARMING_OPTIONS.  But I think we should.

Reply:
Hi @peterbarker yes, I'm sorry but I agree that we should move it.  It's new for 4.6 so I think we could get away without the parameter conversion if we backported it before the official release

Reply:
.... actually, I think the problems revolved around Plane.

If we ever allow Plane to arm without position (@Ryanf55 ...) we would have issues around retaining the default for Plane.  parameter defaults and bitmasks are a mess - if someone already has set ARMING_OPTION in their parameter storage then you can't have sensible default values on a per-vehicle basis (or indeed any non-zero bit default).

Reply:
For copter and rover the default is zero (e.g. require GPS for arming is false) but I guess plane would like that to default to 1.  It's sad to put the option in the wrong place because of this though.  I understand the issue but.. it's hard to accept doing the wrong thing for a small technical reason

Reply:
Good details, I need time to digest, but am swamped at work. If you aren't in a rush, I can discuss in 2 days.

Reply:
PR to move the bit into AP_Arming: https://github.com/ArduPilot/ardupilot/pull/29076